app/ui/browser-modern/actions/page-effects.js

```diff
@@ -17,21 +17,22 @@ import * as EffectTypes from '../constants/effect-types';
 import * as PagesSelectors from '../selectors/pages';
 import PageUIState from '../model/page-ui-state';
 
-export function createPageSession(location, options) {
+export function createPageSession(location, options, id, { withUI } = {}) {
   return {
     type: EffectTypes.CREATE_PAGE_SESSION,
-    id: uuid.v4(),
+    id: id || uuid.v4(),
     location,
     options,
+    withUI,
   };
 }
 
-export function destroyPageSession(pageId) {
+export function destroyPageSession(pageId, { withUI } = {}) {
```

Comment:
We're not defaulting `withUI` here to anything explicitly, but simply destructuring a parameter. The defaulting is done in the sagas (to `true`), not in the effect creators here. I understand the confusion though, so I'll reverse this: default it here and don't do anything special in the sagas.